starfish/spots/_detector/local_max_peak_finder.py

```diff
@@ -297,7 +299,15 @@ def run(
             measurement_function=self.measurement_function,
             radius_is_gyration=False)
 
-        return intensity_table
+        lmr = LocalMaxFinderResults(
+            self._thresholds,
```

Comment:
my understanding is that the self.* variables never get set in the first place when running this w/ multiple processes? i think what we need to verify that this works with a unit test. the test plan is to ensure that the returned lmr indeed has values for the desired variables. 

only thing need to make this work is:
1. a realistic synthetic dataset example
2. a code path that goes through this part of detect_spots:

```
    else:  # don't use a reference image, measure each
        spot_finding_method = partial(spot_finding_method, **spot_finding_kwargs)
        spot_attributes_list = data_stack.transform(
            func=spot_finding_method,
            group_by=group_by
        )
        intensity_table = concatenate_spot_attributes_to_intensities(spot_attributes_list)
```

Reply:
the offending piece of the above code is data_stack.transform, which does this:

```        with multiprocessing.Pool(
                n_processes,
                initializer=SharedMemory.initializer,
                initargs=((self._data._backing_mp_array,
                           self._data._data.shape,
                           self._data._data.dtype),)) as pool:
            results = pool.imap(applyfunc, indices_and_slice_list)
            return list(zip(results, indices))```

Reply:
Ahh this changes things. Wait why does the test need to go through this section of detect spots? `    else:  # don't use a reference image, measure each
        spot_finding_method = partial(spot_finding_method, **spot_finding_kwargs)
        spot_attributes_list = data_stack.transform(
            func=spot_finding_method,
            group_by=group_by
        )
        intensity_table = concatenate_spot_attributes_to_intensities(spot_attributes_list)` I thought that if someone wanted the extra info they would call LocalMaxPeakFinder.run() directly like they do for PIxelSpotDetector in the dartfish notebook? `spot_intensities, results = psd.run(zero_norm_stack)`